storage/local/storage.go

```diff
@@ -360,11 +371,28 @@ func (s *memorySeriesStorage) GetMetricForFingerprint(fp clientmodel.Fingerprint
 
 // AppendSamples implements Storage.
 func (s *memorySeriesStorage) AppendSamples(samples clientmodel.Samples) {
+	if len(samples) == 0 {
```

Comment:
That's an assumption. And this is what I mean by robustness: I don't rely on assumptions about the behavior of the caller.

But if we trust the caller to do the right thing, we can actually remove the mutex altogether, because currently, the method is only called by a single goroutine, and that's by design, not by chance. Without the mutex, the emptiness check is actually superfluous.

I'll do that tomorrow.